addons/stock_account/static/src/stock_account_forecasted/forecasted_header.js

```diff
@@ -0,0 +1,30 @@
+/** @odoo-module **/
+import { patch } from '@web/core/utils/patch';
+
+import { ForecastedHeader as Parent } from "@stock/stock_forecasted/forecasted_header";
+
+export class StockAccountForecastedHeader extends Parent{}
+
+patch(Parent.prototype, 'stock_account.ForecastedHeader', {
+    async _onClickValuation() {
+        const templates = this.props.docs.product_templates_ids;
+        const variants = this.props.docs.product_variants_ids;
+        const context = Object.assign({}, this.context);
+        if (templates) {
+            context.search_default_product_tmpl_id = templates;
+        } else {
+            context.search_default_product_id = variants;
+        }
```

Comment:
Since we have the exact same code in the method `_onClickInventory`, we could imagine to have a common method
```suggestion
        const context = this._getActionContext();
```
(See next comment)